File: src/test/java/utils/JsonReader.java
package utils;

import com.google.common.io.CharStreams;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.fail;

/**
 * Created by vadim on 02.10.16.
 */

public class JsonReader {

    public JSONObject read(String name) throws IOException, NullPointerException {
        JSONObject json = null;
        try {
            json = new JSONObject(readFile(name));
        } catch (IOException | NullPointerException | JSONException e) {
            fail(name + " not found or incorrect");
        }

        assertNotNull("JSON is empty", json);
        return json;
    }

    private String readFile(String name) throws IOException, NullPointerException {
        InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream(name);
        return CharStreams.toString(new InputStreamReader(inputStream, "UTF-8"));
    }
}

File: src/test/kotlin/com/robohorse/robopojogenerator/utils/JsonReader.kt
package com.robohorse.robopojogenerator.utils

import com.google.common.io.CharStreams
import org.json.JSONException
import org.json.JSONObject
import org.junit.Assert
import java.io.IOException
import java.io.InputStreamReader

class JsonReader {

    fun read(name: String): JSONObject? {
        var json: JSONObject? = null
        try {
            json = JSONObject(readFile(name))
        } catch (e: IOException) {
            Assert.fail("$name not found or incorrect")
        } catch (e: NullPointerException) {
            Assert.fail("$name not found or incorrect")
        } catch (e: JSONException) {
            Assert.fail("$name not found or incorrect")
        }
        Assert.assertNotNull("JSON is empty", json)
        return json
    }

    private fun readFile(name: String): String {
        val inputStream = this.javaClass.classLoader.getResourceAsStream(name)
        return CharStreams.toString(InputStreamReader(inputStream, "UTF-8"))
    }
}
